### PR TITLE
Adds WFD attributes to checkin and checkout

### PR DESCRIPTION
**Pull request by bitboxer**
Closes https://github.com/railslove/recover-backlog/issues/137

### CODE REVIEW

File: ui/core/Button.tsx [hunk 1]
```diff
@@ -8,13 +8,21 @@ interface Props {
...
 }
 type ButtonProps = JSX.IntrinsicElements['button'] & Props & As
 
 export const Button: React.FC<ButtonProps> = React.forwardRef(
-  ({ children, left, right, ...rest }, ref) => {
+  ({ children, left, right, dataAttributes, ...rest }, ref) => {
+    const data = {}
+    Object.keys(dataAttributes || {}).forEach(function (key) {
+      data[`data-${key}`] = dataAttributes[key]
+    })
```
Comment (bonflintstone) [inline ui/core/Button.tsx:22]:
```suggestion
      const data = Object.entries(dataAttributes || {})
        .reduce((data, [key, value]) => ({ [`data-${key}`]: value, ...data }), {})
    )
```

Comment (bonflintstone) [reply ui/core/Button.tsx:22]: This would be the functional way to do this in javascript, I'll leave it up to you if this is more or less readable :smile:

**Review by bonflintstone — APPROVED** [review]

**Comment by stringparser** [comment]
Looks good 👍

**Pull request merged**